app/src/main/java/com/cp/tasky/ui/sharecomponent/PasswordTextField.kt

```diff
@@ -0,0 +1,66 @@
+package com.cp.tasky.ui.sharecomponent
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextField
+import androidx.compose.material3.TextFieldDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.tooling.preview.Preview
+import com.cp.tasky.R
+import com.cp.tasky.ui.theme.MEDIUM_8_DP
+
+
+@Composable
+fun PasswordTextField(modifier: Modifier = Modifier) {
+    var password by remember { mutableStateOf("") }
```

Comment:
I would always define the states in a ViewModel, since yours will be reset to an empty string after a screen rotation or any other configuration change. In a ViewModel that can't happen and additionally you strictly separate the UI logic from the logic that decides about how the state looks like.